src/main/java/org/spongepowered/api/text/selector/Selector.java

```diff
@@ -24,57 +24,78 @@
  */
 package org.spongepowered.api.text.selector;
 
+import com.google.common.base.Optional;
 import org.spongepowered.api.entity.Entity;
 import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.extent.Extent;
 
 import java.util.List;
-import java.util.Map;
-
-import javax.annotation.Nullable;
 
 /**
- * Represents an immutable selector. A selector starts with the {@code @} symbol
- * and is followed by a single character signifying which selector is being used
- * (known as the type), and finally the optional arguments in brackets. For
- * example, the all player selector is {@code @a}, and with a radius of 20 it
- * would be {@code @a[r=20]}.
+ * Represents an immutable selector of targets, as used in commands.
+ *
+ * <p>In Vanilla, selectors are mostly represented as plain text, starting with
+ * an {@code @} symbol and followed by a single character signifying the type,
+ * and finally the (optional) arguments in brackets.</p> <p>As an example, the
+ * all player selector is {@code @a}, and with a radius of 20 it would be
+ * {@code @a[r=20]}.</p>
  *
  * @see <a href="http://minecraft.gamepedia.com/Selector#Target_selectors">
- *      TargetSelectors on the Minecraft Wiki</a>
+ *      Target selectors on the Minecraft Wiki</a>
  */
 public interface Selector {
```

Comment:
I'm using the features of github to their fullest potential. Commenting in any other way would be extremely inconvenient and slow.